src/odemis/gui/cont/acquisition.py

```diff
@@ -450,6 +450,455 @@ def open_acquisition_dialog(self):
+        # for the z-stack parameters
+        self._panel.param_Zmin.Bind(wx.EVT_TEXT_ENTER, self._on_Zmin_param)
+        self._panel.param_Zmax.Bind(wx.EVT_TEXT_ENTER, self._on_Zmax_param)
+        self._panel.param_Zstep.Bind(wx.EVT_TEXT_ENTER, self._on_Zstep_param)
+
+        # callbacks of VA's
+        self._tab_data.filename.subscribe(self._on_filename, init=True)
+        self._tab_data.zStackActive.subscribe(self._update_zstack_active, init=True)
+        self._tab_data.streams.subscribe(self._on_streams, init=True)
+        # TODO link .acquiStreams with a callback that is called
+        # to check/uncheck items (?)
+
+        # VA's connector
+        self._zstack_chkbox_vac = VigilantAttributeConnector(
+            va=self._tab_data.zStackActive,
+            value_ctrl=self._panel.z_stack_chkbox,
+            events=wx.EVT_CHECKBOX,
+        )
+
+    def _on_acquire(self, _):
+        """
+        called when the button "acquire" is pressed
+        """
+        # hide/show/disable some widgets
+        self._panel.gauge_cryosecom_acq.Show()
+        self._panel.btn_cryosecom_acqui_cancel.Show()
+        self._panel.txt_cryosecom_left_time.Show()
+        self._panel.txt_cryosecom_est_time.Hide()
+        self._panel.btn_cryosecom_acquire.Disable()
+        self._panel.btn_cryosecom_change_file.Disable()
+        self._panel.btn_acquire_overview.Disable()
+        self._panel.z_stack_chkbox.Disable()
+        self._panel.streams_chk_list.Disable()
+        # disable the streams settings
+        self._tab.streambar_controller.pauseStreams()
+        self._tab.streambar_controller.pause()
+        self._panel.Layout()
+
+        # TODO use the function "generate_zlevels". The zrange
+        # is defined using the zMax and zMin obtained
+        # from the GUI from the user. Similarly for zStep. Pass
+        # the returned dictionary to the acquisition function below,
+        # after replacing "acquire" function with "acquireZStack".
+        self._acq_future = acqmng.acquire(
+            self._acquiStreams.value, self._tab_data.main.settings_obs
+        )
+        self._tab_data.main.is_acquiring.value = True
+        logging.info("Acquisition started")
+        # link the acquisition gauge to the acquisition future
+        self._gauge_future_conn = ProgressiveFutureConnector(
+            future=self._acq_future,
+            bar=self._panel.gauge_cryosecom_acq,
+            label=self._panel.txt_cryosecom_left_time,
+            full=False,
+        )
+        self._acq_future.add_done_callback(self._on_acquisition_done)
+
+    @call_in_wx_main
+    def _on_acquisition_done(self, future):
+        """
+        Called when the acquisition process is
+        done, failed or canceled
+        """
+        self._acq_future = None
+        self._gauge_future_conn = None
+        self._tab_data.main.is_acquiring.value = False
+
+        # try to get the acquisition data
+        try:
+            data, exp = future.result()
```

Comment:
No use for `data, exp` -> `_, _`

Reply:
Actually, you should use it.
If exp is not None, it means not all acquisition succeeded. At least, you should log a error. Possibly, you'd change the Label to "Acquisition partially failed" instead of "Saving file...". 
You could also use `data`: instead of passing `future` to the `export_data`, you'd pass directly `data` and the thumbnail. I think it'd actually help to make the code clearer too.